wandb/sdk/launch/runner/kubernetes_runner.py

```diff
@@ -681,7 +682,9 @@ def maybe_create_imagepull_secret(
         A secret if one was created, otherwise None.
     """
     secret = None
-    if isinstance(registry, LocalRegistry):
+    if isinstance(registry, LocalRegistry) or isinstance(
+        registry, AzureContainerRegistry
```

Comment:
This is surprising, if a user did not want to set up an Azure AD workload identity, would this be necessary?

If so we may want to add that as an option? The docs here seem to make indicate that this is possible: https://learn.microsoft.com/en-us/azure/container-registry/container-registry-auth-kubernetes#create-an-image-pull-secret

Also, what if a user wants to use an Azure Container Registry outside of AKS (kind of like our support for ECR outside of EKS) would this be necessary then?

Reply:
I think we should make like a matrix of our cloud support indicating how each of them work, and showing them in the launch readme.

Reply:
Yeah this PR is only introducing support for AKS with workload identity, not support for using ACR and blob storage from a local cluster. We could allow a user to store their username/password for the registry in a k8s secret and then use that as the pull secret. I'm going to make a ticket for that here:
https://wandb.atlassian.net/browse/WB-14052